localstack/services/stepfunctions/asl/component/state/state_execution/state_parallel/branches_decl.py

```diff
@@ -1,41 +1,116 @@
-from typing import Final
+import copy
+import datetime
+import threading
+from typing import Final, Optional
 
+from localstack.aws.api.stepfunctions import ExecutionFailedEventDetails, HistoryEventType
+from localstack.services.stepfunctions.asl.component.common.error_name.custom_error_name import (
+    CustomErrorName,
+)
+from localstack.services.stepfunctions.asl.component.common.error_name.failure_event import (
+    FailureEvent,
+    FailureEventException,
+)
 from localstack.services.stepfunctions.asl.component.eval_component import EvalComponent
 from localstack.services.stepfunctions.asl.component.program.program import Program
-from localstack.services.stepfunctions.asl.eval.count_down_latch import CountDownLatch
+from localstack.services.stepfunctions.asl.component.state.state_execution.state_parallel.branch_worker import (
+    BranchWorker,
+)
 from localstack.services.stepfunctions.asl.eval.environment import Environment
-from localstack.services.stepfunctions.asl.eval.program_worker import ProgramWorker
+from localstack.services.stepfunctions.asl.eval.event.event_detail import EventDetails
+from localstack.services.stepfunctions.asl.eval.program_state import ProgramError, ProgramState
+from localstack.utils.collections import select_from_typed_dict
+
+
+class BranchWorkerPool(BranchWorker.BranchWorkerComm):
+    _mutex: Final[threading.Lock]
+    _termination_event: Final[threading.Event]
+    _active_workers_num: int
+
+    _terminated: bool
```

Comment:
isn't this equivalent to checking if the `_termination_event` is set?

Reply:
Indeed, thank you!